relatedness2.py

```diff
@@ -0,0 +1,99 @@
+
+    # Check to see if there are the correct amount of columns in each file and if yes then rename column names in ped file
+    if len(relateFile.columns) != 7:
+        return False, 'There are the incorrect amount of columns within the relatedness file'
+    else:
+        pass
+
+    #### This commented section creates a unique list of family IDs from the ped file, therefore it works with multiple families
+    if len(pedFile.columns) != 6:
+        return False, 'There are the incorrect amount of columns within the pedigree file'
+    else:
+        pedFile.columns = ['familyid', 'individualid', 'paternalid', 'maternalid', 'sex', 'phenotype']
+    
+    familyIdentifier = pedFile.familyid.unique().tolist()
+    familyIdentifier.remove('0')
+
+    # Creating an empty dictionary to store family identifiers and also their variables (key value pairs)
+    familyID = {}
+
+    for ID in familyIdentifier:
+        familyID[ID] = {}
+
+    #This will assign the mum for each ID! crack on with this tomorrow!
+    for ID in familyIdentifier:                                                                      
+       for x in pedFile.index:                                                                       
+            if pedFile.paternalid[x] == '0' and pedFile.sex[x] == 2 and pedFile.familyid[x] == ID:
+                familyID[ID]['mum'] = pedFile.individualid[x]
+            elif pedFile.paternalid[x] == '0' and pedFile.sex[x] == 1 and pedFile.familyid[x] == ID:
+                familyID[ID]['dad'] = pedFile.individualid[x]
+            elif pedFile.paternalid[x] != '0' and pedFile.sex[x] != 0 and pedFile.familyid[x] == ID:
+                familyID[ID]['proband'] = pedFile.individualid[x]
+            if pedFile.sex[x] > 2 or pedFile.phenotype[x] > 2:
+                return False
+
+
+    # Getting rid of the NTCs in INDV columns
+    noNTC = relateFile[relateFile['INDV1'] != 'NTC']
+    relateFile = noNTC[noNTC['INDV2'] != 'NTC']
+
+
+    for ID in familyIdentifier:
+        for x in relateFile.index:
+            if relateFile['INDV1'][x] == familyID[ID]['proband'] or relateFile['INDV2'][x] == familyID[ID]['proband']:
+                if relateFile['INDV1'][x] == familyID[ID]['mum'] or relateFile['INDV1'][x] == familyID[ID]['dad'] or relateFile['INDV2'][x] == familyID[ID]['mum'] or relateFile['INDV2'][x] == familyID[ID]['dad']:
+                    if relateFile['RELATEDNESS_PHI'][x] >= 0.2 and relateFile['RELATEDNESS_PHI'][x] <= 0.3:
+                        pass
+                    elif relateFile['RELATEDNESS_PHI'][x] < 0.2 or relateFile['RELATEDNESS_PHI'][x] > 0.3:
+                        return False
+            if relateFile['INDV1'][x] == familyID[ID]['mum'] and relateFile['INDV2'][x] == familyID[ID]['dad'] and relateFile['RELATEDNESS_PHI'][x] >= 0.04:
+                return False, 'Looks like a possible problem. Check to see if parents are related'
+        return True
+
```

Comment:
no return at bottom of cuntion causing errors I think - see extra tests I added.